htdocs/admin/content/messages.php

```diff
@@ -1,822 +1,714 @@
-
-			$tblcnt = mysql_fetch_row(dbquery("SELECT count(*) FROM messages;"));
-			echo "<br/>Es sind ".nf($tblcnt[0])." Eintr&auml;ge in der Datenbank vorhanden.";
-		}
-
-	}
-?>
-
+use EtoA\Admin\AdminUser;
+use EtoA\Message\MessageCategoryRepository;
+use EtoA\Message\MessageRepository;
+use EtoA\Message\ReportRepository;
+use EtoA\User\UserRepository;
+use Symfony\Component\HttpFoundation\Request;
+
+define("USER_MESSAGE_CAT_ID",1);
+define("SYS_MESSAGE_CAT_ID",5);
+
+/** @var \EtoA\User\UserRepository $userRepository */
+$userRepository = $app['etoa.user.repository'];
+
+/** @var MessageRepository */
+$messageRepository = $app[MessageRepository::class];
+
+/** @var MessageCategoryRepository */
+$messageCategoryRepository = $app[MessageCategoryRepository::class];
+
+/** @var ReportRepository */
+$reportRepository = $app[ReportRepository::class];
+
+/** @var Request */
+$request = Request::createFromGlobals();
+
+if ($sub=="sendmsg") {
+    sendMessageForm($request, $cu, $messageRepository, $userRepository);
+} elseif ($sub=="reports") {
+    manageReports($request, $reportRepository, $userRepository);
+} else {
+    manageMessages($request, $messageRepository, $messageCategoryRepository, $userRepository);
+}
+
+function sendMessageForm(
+    Request $request,
+    AdminUser $cu,
+    MessageRepository $messageRepository,
+    UserRepository $userRepository
+): void {
+    global $page;
+    global $sub;
+
+    echo "<h1>Nachrichten</h1>";
+
+    echo "Nachricht an einen Spieler senden:<br/><br/>";
+
+    $subj = $request->query->get('message_subject', '');
+    $text = "";
+    if ($request->request->has('submit'))
+    {
+        if ($request->request->get('message_subject') != "" && $request->request->get('message_text') != "")
+        {
+            $to = [];
+            if ($request->request->getInt('rcpt_type') == 1) {
+                $to = $userRepository->getEmailAddresses();
+            } else {
+                $userId = $request->request->getInt('message_user_to');
+                $recipient = $userRepository->getUser($userId);
+                $to[$userId] = $recipient->getEmailAddress();
+            }
+
+            $msg_type = $request->request->getInt('msg_type');
+
+            if ($msg_type==1 || $msg_type==2)
+            {
+                $mail = new Mail($request->request->get('message_subject'), $request->request->get('message_text'));
+                if ($request->request->getInt('from_id') > 0)
+                {
+                    $replyUser = $userRepository->getUser($cu->playerId);
+                    $reply = $replyUser->getEmailAddress();
+                }
+                else
+                {
+                    $reply = "";
+                }
+            }
+
+            $mailCnt = 0;
+            $msgCnt = 0;
+
+            foreach ($to as $k=>$v)
+            {
+                if ($msg_type==0 || $msg_type==2)
```

Comment:
`===` or `in_array([0,2], $msg_type)`

even better: extract `msg_type` in an enum

Reply:
done